### PR TITLE
Add pgAudit extension

### PR DESCRIPTION
**Pull request by to-bar**

### CODE REVIEW

File: core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/configure-postgresql.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
+---
+- name: Set OS specific paths
+  set_fact:
+    pg_config_dir: "{{ pg_config_dir[ansible_os_family] }}"
+  vars:
+    pg_config_dir:
```
Comment (erzetpe) [inline core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/configure-postgresql.yml]: Cannot this be moved to defaults yaml for PostgreSQL role?

File: core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/configure-postgresql.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    path: "{{ pg_config_dir }}/pg_hba.conf"
+    regexp: '^host.*?all.*?all.*?127\.0\.0\.1\/32.*?ident$'
+    replace: 'host    all             all       0.0.0.0/0          md5'
+    backup: yes
+  register: change_pg_hba_conf
+
+- name: Configure pgAudit extension
+  include_tasks: extensions/configure-pgaudit.yml
+  when:
+    - specification.extensions.pgaudit.enabled is defined
```
Comment (erzetpe) [inline core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/configure-postgresql.yml]: What if pgaudit will be undefined?

File: core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/configure-postgresql.yml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    name: postgresql
+    state: restarted
+  when: change_pg_hba_conf
+     or change_postgresql_epiphany_conf
+     or change_postgresql_conf
+
+- name: Set up replication
+  include_tasks: replication-{{ ansible_os_family }}.yml
+  when:
+    - specification.replication.enable is defined
```
Comment (erzetpe) [inline core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/configure-postgresql.yml]: What if replication will not be defined?

File: core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/install-packages/Debian.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    update_cache: yes
+    state: present
+
+- name: Install pgAudit package
+  apt:
+    name:
+      - postgresql-10-pgaudit
+    state: present
+  when:
+    - specification.extensions.pgaudit.enabled is defined
```
Comment (erzetpe) [inline core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/install-packages/Debian.yml]: What if pgaudit not defined?

File: core/src/epicli/data/common/ansible/playbooks/roles/postgresql/templates/postgresql-epiphany.conf.j2 [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+{%         if value != 'SET_BY_AUTOMATION' %}
+{{ parameter }} = {{ value }}
+{%         else %}
+{# pgaudit must be loaded in shared_preload_libraries #}
+{%           if extension == 'pgaudit' and parameter == 'shared_preload_libraries' %}
+{%             if runtime_shared_preload_libraries is defined and (runtime_shared_preload_libraries|length) %}
+shared_preload_libraries = '{{ (runtime_shared_preload_libraries.split(',') + ['pgaudit']) | unique | join(',') }}'
+{%             else %}
+shared_preload_libraries = 'pgaudit'
+{%             endif %}
```
Comment (erzetpe) [inline core/src/epicli/data/common/ansible/playbooks/roles/postgresql/templates/postgresql-epiphany.conf.j2]: Here we need logic, that will support also additional modules.

File: core/src/epicli/data/common/defaults/configuration/postgresql.yml [hunk 6]
```diff
@@ -1,7 +1,38 @@
...
+        subgroups:
+          - name: Where to Log
+            parameters:
+              - name: log_filename
+                value: "'postgresql-%Y-%m-%d_%H%M%S.log'"
+  extensions:
+    pgaudit:
+      enabled: true
+      config_file_parameters:
+        shared_preload_libraries: SET_BY_AUTOMATION
```
Comment (erzetpe) [inline core/src/epicli/data/common/defaults/configuration/postgresql.yml]: This also should be defined on per module basis and not set by automation.

File: core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/configure-postgresql.yml (hunk 3 above)
Comment (to-bar) [reply core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/configure-postgresql.yml]: Then condition will be `false` and task will be skipped.

File: core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/configure-postgresql.yml (hunk 2 above)
Comment (to-bar) [reply core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/configure-postgresql.yml]: Then condition will be `false` and task will be skipped.

File: core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/install-packages/Debian.yml (hunk 4 above)
Comment (to-bar) [reply core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/install-packages/Debian.yml]: Then condition will be `false` and task will be skipped.

File: core/src/epicli/data/common/defaults/configuration/postgresql.yml (hunk 6 above)
Comment (to-bar) [reply core/src/epicli/data/common/defaults/configuration/postgresql.yml]: Done

File: core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/configure-postgresql.yml (hunk 1 above)
Comment (to-bar) [reply core/src/epicli/data/common/ansible/playbooks/roles/postgresql/tasks/configure-postgresql.yml]: I don't think that adding rather "static" options (paths depend on installed packages) to the configuration is a good idea unless someone needs them.

File: core/src/epicli/data/common/ansible/playbooks/roles/postgresql/templates/postgresql-epiphany.conf.j2 (hunk 5 above)
Comment (to-bar) [reply core/src/epicli/data/common/ansible/playbooks/roles/postgresql/templates/postgresql-epiphany.conf.j2]: Done

**Pull request merged**